test_echo.py

```diff
@@ -1,13 +1,20 @@
 from echo_client import client
 
 
-def test_1():
-    assert client('This is a unicode test') == 'This is a unicode test'
+def test_unicode():
+    assert client(u'This is a unicode test'.encode('utf-8')) == \
+        u'This is a unicode test'.decode('utf-8')
```

Comment:
a few things here.  A unicode string that doesn't contain non-ASCII characters is not particularly useful.  All the problems arise when you try to convert non-ascii chars.  Second, calling `decode` on a unicode object is exactly the type of acton that results in errors.  A unicode object is already `decoded`.  It doesn't need to be decoded, and when you try to do so, Python will helpfully encode it first, using sys.defaultencoding (usually ascii).  Boom.